client/components/count/index.jsx

```diff
@@ -2,22 +2,22 @@
  * External dependencies
  */
 import React from 'react';
-import PureRenderMixin from 'react-pure-render/mixin';
+import { localize } from 'i18n-calypso';
+import { identity, omit } from 'lodash';
 
-export default React.createClass( {
+const Count = ( { count, numberFormat, ...inheritProps } ) => (
+	<span className="count" { ...omit( inheritProps, [ 'translate', 'moment' ] ) }>
+		{ numberFormat( count ) }
+	</span>
+);
 
-	displayName: 'Count',
+Count.propTypes = {
+	count: React.PropTypes.number.isRequired,
+	numberFormat: React.PropTypes.func
+};
 
-	mixins: [ PureRenderMixin ],
+Count.defaultProps = {
+	numberFormat: identity
```

Comment:
Yep, I agree. The unit in this case is just the component, and we can rely on `i18n-calypso`'s [own tests](https://github.com/Automattic/i18n-calypso/blob/711ff83895de234ffa06a6b14551d11be48817b6/test/index.js#L216-L241) for proper behavior of `numberFormat` -- it's not our concern here.

So we should really just test if `Count` renders a `span` with the correct `className`, and if it calls a given `numberFormat` prop function on its `count` prop. This is probably best done using `enzyme`' [`shallow`](https://github.com/airbnb/enzyme/blob/ea8be1701049d3b1e48a6fd6b438e7834cfbdaa3/docs/api/shallow.md) testing util (instead of `react-addons-test-utils`).